lib/data_source/client/github.rb

```diff
@@ -7,40 +7,105 @@
 module DataSource
   module Client
     class Github
-      CACHE_TTL_SECS = 3600 * 12
 
       def initialize(config)
         @host = config[:host]
         @access_token = config[:access_token]
         @cache_dir = config[:cache_dir]
+        @me_account = config[:me_account]
+        @pr_all_involve_me = config[:pr_all_involve_me]
+
+        @cache_name_hash = {
+            user_repos: 'user_repos',
+            user_orgs: 'user_orgs',
+            user_pulls: 'user_pulls'
+        }
+
+        @cache_ttl_hash = {
+            user_repos: config[:cache_ttl_sec_repo],
+            user_orgs: config[:cache_ttl_sec_org],
+            user_pulls: config[:cache_ttl_sec_pr]
+        }
       end
 
       def search_repos(params)
         modifiers = ['in:name']
-        modifiers += org_modifiers('user:@me') if params[:mine]
+        modifiers += org_modifiers("user:#@me_account") if params[:mine]
         params = search_params(params[:query], modifiers)
         response = request('/search/repositories', params)
         handle_response(response)[:items]
       end
 
       def user_repos
         params = {
-          sort: 'pushed',
-          direction: 'desc',
-          per_page: 100
+            sort: 'pushed',
+            direction: 'desc',
+            per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        with_cache(:user_repos) do
+
+          page_count = get_total_page_for_request('/user/repos', params)
+          if page_count != nil
+            all_user_repos = Array.new
+            write = true
+            (1..page_count).step(1) do |n|
+              params[:page] = n
+              part_res = request('/user/repos', params)
+              if not part_res.is_a?(Net::HTTPSuccess)
+                write = false
+                break
+              else
+                all_user_repos = all_user_repos+ deserialize_body(part_res.body)
+              end
+            end
+            {write: write, cache_content: all_user_repos}
```

Comment:
now the cache component interface is taking object as argument instead of string.